### ISSUE TITLE
Some question about train STN 

### ISSUE THREAD

**Issue by xywlpo** [charigyang/itsabouttime#4]
Hi, thank you for your amazing work. I am inspired by your work about alignment the clock. I have tried to use STN for alignment my pointer meter as followings. I picked front meter of each category as a standard meter. And online random homography matrix was generated by code to apply on these standard meters for training STN. My batchsize is 32, and iteration about 120000,  trian data is 10000, but the results is not good, and the loss is about 0.5 finally. Could you give my some advice about this task?

Thanks very much.
![circle29](https://user-images.githubusercontent.com/728405/174542165-84df1b01-f8b2-4873-8a67-c8eb4d1a13e1.jpg)
![circle48](https://user-images.githubusercontent.com/728405/174542192-a19d0af4-7f9b-447f-a453-87ebdcbbf6ae.jpg)
![circle129](https://user-images.githubusercontent.com/728405/174542236-ece6da77-3a0f-42fb-9c43-7a0fb35baec4.jpg)

**Comment by charigyang** [comment]
Hi thanks for your interest, we supervise the STN using synthetic data (i.e. when we generate them we also have GT parameters for the STN). We find that that helps. I suppose if you use our pretrained model for STN it might be not too bad.

**Comment by xywlpo** [comment]
> Hi thanks for your interest, we supervise the STN using synthetic data (i.e. when we generate them we also have GT parameters for the STN). We find that that helps. I suppose if you use our pretrained model for STN it might be not too bad.

Thanks for your reply. I use your pretrained model full+++_st.pth as my initial model weights. If I just use synthetic data to train the model, do you think that will be fine? 
Could you tell me your loss value when the model training finished?
Did you just use synthetic data to train your STN without any real data ?

**Comment by charigyang** [comment]
STN is trained with synthetic as supervision (though the gradient should flow end-to-end when training recognition). I think we trained synthetic one fine though. Loss ends around 7*10^-3 (for STN) and 0.5 (for recognition).

**Comment by xywlpo** [comment]
> flow

Thanks very much. I will try to generate synthetic data to simulate my real meter, and then to train my model.

**Comment by sajsxj** [comment]
> > flow
> 
> Thanks very much. I will try to generate synthetic data to simulate my real meter, and then to train my model.

hi, what was the result and could u share u approach? thank u very much!

**Comment by sajsxj** [comment]
> > flow
> 
> Thanks very much. I will try to generate synthetic data to simulate my real meter, and then to train my model.

hi, what was the result and could u share ur approach? thank u very much!
